```
======================================================================
ERROR: custom.intrahealth.tests.reports.test_recap_passage_report_v2:TestRecapPassageReport2.test_recap_passage_report
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/intrahealth/tests/reports/test_recap_passage_report_v2.py", line 32, in test_recap_passage_report
    recap_passage_report2_second = recap_passage_report2.report_context['reports'][1]['report_table']
IndexError: list index out of range
----------------------------------------------------------------------
```

--- a/custom/intrahealth/reports/recap_passage_report_v2.py
+++ b/custom/intrahealth/reports/recap_passage_report_v2.py
@@ -2,10 +2,10 @@ from __future__ import absolute_import
 from __future__ import unicode_literals
 from memoized import memoized
 
-from corehq.apps.reports.standard import MonthYearMixin
 from custom.intrahealth.filters import YeksiNaaLocationFilter2, FRMonthFilter, FRYearFilter
 from custom.intrahealth.sqldata import RecapPassageData2, DateSource2
 from custom.intrahealth.reports.tableu_de_board_report_v2 import MultiReport
+from custom.intrahealth.reports.utils import MonthYearMixin
 
 
 class RecapPassageReport2(MonthYearMixin, MultiReport):

--- a/custom/intrahealth/reports/recap_passage_report_v2.py
+++ b/custom/intrahealth/reports/recap_passage_report_v2.py
@@ -5,10 +5,10 @@ from memoized import memoized
 from custom.intrahealth.filters import YeksiNaaLocationFilter2, FRMonthFilter, FRYearFilter
 from custom.intrahealth.sqldata import RecapPassageData2, DateSource2
 from custom.intrahealth.reports.tableu_de_board_report_v2 import MultiReport
-from custom.intrahealth.reports.utils import MonthYearMixin
+from custom.intrahealth.reports.utils import YeksiNaaMonthYearMixin
 
 
-class RecapPassageReport2(MonthYearMixin, MultiReport):
+class RecapPassageReport2(YeksiNaaMonthYearMixin, MultiReport):
     title = "Recap Passage NEW"
     name = "Recap Passage NEW"
     slug = 'recap_passage2'

--- a/custom/intrahealth/reports/utils.py
+++ b/custom/intrahealth/reports/utils.py
@@ -6,13 +6,12 @@ from datetime import datetime
 from corehq.apps.products.models import SQLProduct
 from corehq.apps.locations.models import get_location
 from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumnGroup, DataTablesColumn
-from corehq.apps.reports.filters.select import MonthFilter, YearFilter
+from corehq.apps.reports.standard import MonthYearMixin
 from corehq.apps.reports.sqlreport import DataFormatter, DictDataFormat
 from corehq.util.translation import localize
 from custom.intrahealth.sqldata import NombreData, TauxConsommationData
 from django.utils.translation import ugettext as _
 from memoized import memoized
-from dimagi.utils.dates import DateSpan
 from dimagi.utils.parsing import json_format_date
 from six.moves import zip
 from six.moves import range
@@ -25,23 +24,7 @@ def get_localized_months():
         return [(_(calendar.month_name[i])).title() for i in range(1, 13)]
 
 
-class MonthYearMixin(object):
-    """
-        Similar to DatespanMixin, but works with MonthField and YearField
-    """
-    fields = [MonthFilter, YearFilter]
-
-    _datespan = None
-
-    @property
-    def datespan(self):
-        if self._datespan is None:
-            datespan = DateSpan.from_month(self.month, self.year)
-            self.request.datespan = datespan
-            self.context.update(dict(datespan=datespan))
-            self._datespan = datespan
-        return self._datespan
-
+class YeksiNaaMonthYearMixin(MonthYearMixin):
     @property
     def month(self):
         if 'month' in self.request.GET:

--- a/custom/intrahealth/reports/utils.py
+++ b/custom/intrahealth/reports/utils.py
@@ -2,14 +2,17 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 import calendar
+from datetime import datetime
 from corehq.apps.products.models import SQLProduct
 from corehq.apps.locations.models import get_location
 from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumnGroup, DataTablesColumn
+from corehq.apps.reports.filters.select import MonthFilter, YearFilter
 from corehq.apps.reports.sqlreport import DataFormatter, DictDataFormat
 from corehq.util.translation import localize
 from custom.intrahealth.sqldata import NombreData, TauxConsommationData
 from django.utils.translation import ugettext as _
 from memoized import memoized
+from dimagi.utils.dates import DateSpan
 from dimagi.utils.parsing import json_format_date
 from six.moves import zip
 from six.moves import range
@@ -20,6 +23,38 @@ def get_localized_months():
     #Returns chronological list of months in french language
     with localize('fr'):
         return [(_(calendar.month_name[i])).title() for i in range(1, 13)]
+
+
+class MonthYearMixin(object):
+    """
+        Similar to DatespanMixin, but works with MonthField and YearField
+    """
+    fields = [MonthFilter, YearFilter]
+
+    _datespan = None
+
+    @property
+    def datespan(self):
+        if self._datespan is None:
+            datespan = DateSpan.from_month(self.month, self.year)
+            self.request.datespan = datespan
+            self.context.update(dict(datespan=datespan))
+            self._datespan = datespan
+        return self._datespan
+
+    @property
+    def month(self):
+        if 'month' in self.request.GET:
+            return int(self.request.GET['month'])
+        else:
+            return datetime.utcnow().month
+
+    @property
+    def year(self):
+        if 'year' in self.request.GET:
+            return int(self.request.GET['year'])
+        else:
+            return datetime.utcnow().year
 
 
 class IntraHealthLocationMixin(object):

--- a/custom/intrahealth/tests/reports/test_recap_passage_report_v2.py
+++ b/custom/intrahealth/tests/reports/test_recap_passage_report_v2.py
@@ -17,11 +17,11 @@ class TestRecapPassageReport2(YeksiTestCase):
         mock = MagicMock()
         mock.couch_user = self.user
         mock.GET = {
-            'startdate': '2014-06-01',
-            'enddate': '2014-07-31',
+            'month': '07',
+            'year': '2014',
             'location_id': '1991b4dfe166335e342f28134b85fcac',
         }
-        mock.datespan = DateSpan(datetime.datetime(2014, 6, 1), datetime.datetime(2014, 7, 31))
+        mock.datespan = DateSpan(datetime.datetime(2014, 6, 1), datetime.datetime(2014, 6, 30))
 
         recap_passage_report2 = RecapPassageReport2(request=mock, domain='test-pna')
 
@@ -29,10 +29,9 @@ class TestRecapPassageReport2(YeksiTestCase):
         headers = recap_passage_report2_first['headers'].as_export_table[0]
         rows = recap_passage_report2_first['rows']
         title = recap_passage_report2_first['title']
-        recap_passage_report2_second = recap_passage_report2.report_context['reports'][1]['report_table']
         self.assertEqual(
             title,
-            'Recap Passage 2014-06-10'
+            'Recap Passage 2014-07-22'
         )
         self.assertEqual(
             headers,
@@ -46,45 +45,14 @@ class TestRecapPassageReport2(YeksiTestCase):
         self.assertEqual(
             rows,
             [
-                ['CU', 12, 12, 0, 12, 0, 0, 0, 0, 0, 12, 0, 0],
-                ['Collier', 9, 9, 0, 9, 4, 0, 0, 0, 0, 9, 4, 0],
-                ['DIU', 21, 7, 30, 37, 0, 0, 0, 14, 14, 37, -14, 0],
-                ['Depo-Provera', 791, 606, 0, 606, 0, 0, 0, 185, 185, 606, -185, 0],
-                ['Jadelle', 45, 28, 20, 48, 0, 0, 0, 17, 17, 48, -17, 0],
-                ['Microgynon/Lof.', 801, 522, 160, 682, 0, 0, 0, 279, 279, 682, -279, 0],
-                ['Microlut/Ovrette', 234, 197, 0, 197, 0, 0, 0, 37, 37, 197, -37, 0],
-                ['Preservatif Feminin', 20, 10, 10, 20, 0, 0, 0, 10, 10, 20, -10, 0],
-                ['Preservatif Masculin', 279, 258, 100, 358, 68, 0, 21, 0, 21, 358, 47, 0],
-            ]
-        )
-        headers = recap_passage_report2_second['headers'].as_export_table[0]
-        rows = recap_passage_report2_second['rows']
-        title = recap_passage_report2_second['title']
-        self.assertEqual(
-            title,
-            'Recap Passage 2014-06-11'
-        )
-        self.assertEqual(
-            headers,
-            [
-                'Designations', 'Stock apres derniere livraison',
-                'Stock disponible et utilisable a la livraison',
-                'Livraison', 'Stock Total', 'Precedent', 'Recu hors entrepots mobiles', 'Non Facturable',
-                'Facturable', 'Reelle', 'Stock Total', 'PPS Restant', 'Pertes et Adjustement'
-            ]
-        )
-        self.assertEqual(
-            rows,
-            [
-                ['CU', 10, 9, 0, 9, 0, 0, 0, 1, 1, 9, -1, 0],
-                ['Collier', 6, 6, 0, 6, 0, 0, 0, 0, 0, 6, 0, 0],
-                ['DIU', 6, 6, 0, 6, 0, 0, 0, 0, 0, 6, 0, 0],
-                ['Depo-Provera', 160, 145, 0, 145, 0, 0, 0, 15, 15, 145, -15, 0],
-                ['Jadelle', 13, 13, 0, 13, 0, 0, 0, 0, 0, 13, 0, 0],
-                ['Microgynon/Lof.', 189, 156, 0, 156, 0, 0, 0, 33, 33, 156, -33, 0],
-                ['Microlut/Ovrette', 100, 88, 18, 106, 0, 0, 0, 12, 12, 106, -12, 0],
-                ['Preservatif Feminin', 15, 15, 0, 15, 0, 0, 0, 0, 0, 15, 0, 0],
-                ['Preservatif Masculin', 160, 160, 0, 160, 0, 0, 0, 0, 0, 160, 0, 0],
+                ['CU', 3, 3, 2, 5, 0, 0, 0, 0, 0, 5, 0, 0],
+                ['DIU', 2, 2, 3, 5, 0, 0, 0, 0, 0, 5, 0, 0],
+                ['Depo-Provera', 60, 52, 25, 77, 0, 0, 0, 8, 8, 77, -8, 0],
+                ['Jadelle', 6, 6, 5, 11, 0, 0, 0, 0, 0, 11, 0, 0],
+                ['Microgynon/Lof.', 42, 42, 0, 42, 0, 0, 0, 0, 0, 42, 0, 0],
+                ['Microlut/Ovrette', 15, 12, 18, 30, 0, 0, 0, 3, 3, 30, -3, 0],
+                ['Preservatif Feminin', 5, 5, 0, 5, 0, 0, 0, 0, 0, 5, 0, 0],
+                ['Preservatif Masculin', 100, 100, 0, 100, 0, 0, 0, 0, 0, 100, 0, 0]
             ]
         )
 
@@ -92,11 +60,11 @@ class TestRecapPassageReport2(YeksiTestCase):
         mock = MagicMock()
         mock.couch_user = self.user
         mock.GET = {
-            'startdate': '2014-06-01',
-            'enddate': '2014-07-31',
+            'month': '06',
+            'year': '2014',
             'location_id': '',
         }
-        mock.datespan = DateSpan(datetime.datetime(2014, 6, 1), datetime.datetime(2014, 7, 31))
+        mock.datespan = DateSpan(datetime.datetime(2014, 6, 1), datetime.datetime(2014, 6, 30))
 
         recap_passage_report2 = RecapPassageReport2(request=mock, domain='test-pna')
 

